Tropos/Controllers/TRAppDelegate.m

```diff
@@ -38,6 +44,35 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     return YES;
 }
 
+- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken;
+{
+    NSString *channel = [[NSTimeZone defaultTimeZone] name];
+    [self.courier subscribeToChannel:channel withToken:deviceToken];
+}
+
+- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(nonnull void (^)(UIBackgroundFetchResult))completionHandler;
+{
+    if (![userInfo valueForKeyPath:@"aps.content-available"]) {
+        return;
+    }
+    TRWeatherController *controller = [TRWeatherController new];
+
+    [controller.conditionsDescription subscribeNext:^(NSAttributedString *conditions) {
+        UILocalNotification *notification = [[UILocalNotification alloc] init];
+        notification.fireDate = [NSDate distantPast];
+        notification.alertTitle = @"Today's Weather Forecast";
+        notification.alertBody = conditions.string;
+        [application scheduleLocalNotification:notification];
+
+        completionHandler(UIBackgroundFetchResultNewData);
+
+    } error:^(NSError *error) {
+        completionHandler(UIBackgroundFetchResultFailed);
+    }];
+
+    [controller.updateWeatherCommand execute:self];
+}
```

Comment:
I'm thinking that the bulk of this logic might be better moved to `TRApplicationController`, to keep the application delegate method small. Similar to the background fetch scenario.

That would also mean that you can reuse the same weather controller instance. The main reason I think this is beneficial is that the internal `RACCommand` inside the weather controller ensures that only one weather update is ever firing at a time. But if you have two separate instances, you can technically have multiple requests running concurrently (e.g., if a push notification is received while the app is already in the foreground).